### DISCUSSION TITLE
MPLS

### DISCUSSION THREAD

**Discussion by barajus** [bgplab/bgplab#18] [Ideas]
Hi Ivan, 
I have been watching your _MPLS Essentials_ webinar, and logically, since the content is now free, having MPLS labs, a bit like the BGP series, from easy to expert level, would be amazing :)

**Comment by ipspace** [comment]
Got the message ;) The "only" problem is how to make them easy-to-consume. You cannot load kernel modules in Codespaces (for obvious reasons), so we can't use FRR or VyOS.

Maybe I could work around the Arista cEOS limitations and get MPLS-over-VLAN-interfaces on cEOS to work in Codespaces, otherwise the minimum requirements would be a Ubuntu VM with FRR containers. Anyway, the MPLS data plane is a beast to configure on FRR.

Unfortunately, I have no idea how many people are willing to invest their time into setting up netlab infrastructure. It looks like the BGP labs got any real traction only when I found out about Codespaces.

**Reply by barajus** [reply]
I see, that's sad, clab + FRR is a good combo for those kind of labs. 
I don't see either any workaround for this as the kernel module is a must indeed. 
Hopefully at some point Codespaces will be more flexible with this kind of use-cases..

**Comment by barajus** [comment]
I have the feeling this conversation will be closed soon 😄 
Nice to finally have an option for MPLS in Codespaces

**Comment by ipspace** [comment]
Yeah, now I can start working on those labs ;)) 🎉🎉
